shell/src/main/java/alluxio/cli/fs/command/FreeCommand.java

```diff
@@ -64,6 +68,15 @@ protected void runPlainPath(AlluxioURI path, CommandLine cl)
       throws AlluxioException, IOException {
     FreeOptions options = FreeOptions.defaults().setRecursive(true).setForced(cl.hasOption("f"));
     mFileSystem.free(path, options);
+    CommonUtils.waitFor("File to be freed.", (v) -> {
+      try {
+        return mFileSystem.getStatus(path).getInMemoryPercentage() == 0;
+      } catch (Exception e) {
+        Throwables.propagateIfPossible(e);
+        throw new RuntimeException(e);
+      }
+    }, WaitForOptions.defaults().setTimeoutMs(10 * Constants.SECOND_MS)
```

Comment:
should this be calculated based on the conf of worker heartbeat interval ? by default it is 1sec, but it may be much higher for some settings? I feel it is safer to derive the time out  based on conf rather than a constant based on default value.

Reply:
Updated, thanks.